chips/earlgrey/src/lib.rs

```diff
@@ -8,6 +8,7 @@
 #![no_std]
 #![crate_name = "earlgrey"]
 #![crate_type = "rlib"]
+#![recursion_limit = "256"]
```

Comment:
```suggestion
// `registers/rv_plic_regs` has many register definitions in `register_structs()!`
// and requires a deeper recursion limit than the default to fully expand.
#![recursion_limit = "256"]
```

Reply:
Done.  I don't particularly like what the register generator did for the PLIC registers.  I'm leaning towards declaring the PLIC to be an exception to auto-generated registers.

The problem is that the source-of-truth JSON description of the registers individually enumerates all of the `PRIOn` registers when clearly it should be though of as a sequence of length `n` of identical registers.

If we make an exception for the PLIC regs, we can evict `rv_plic_regs.rs` from the codebase and remove the change to the recursion limit.  We can make that decision when we work on converting the individual peripheral/driver implementations over to the auto-generated definitions.